src/components/Korrel8rPanel.tsx

```diff
@@ -0,0 +1,93 @@
+import { Korrel8rTopology } from './Korrel8rTopology';
+import './korrel8rpanel.css';
+import { getNeighborsGraph } from '../korrel8r-client';
+import { useDispatch, useSelector } from 'react-redux';
+import { setQueryResponse, setQuery } from '../redux-actions';
+import { State } from '../redux-reducers';
+import { Korrel8rGraphNeighboursResponse } from '../korrel8r/query.types';
+import { LoadingTopology } from './LoadingTopology';
+
+export default function Korrel8rPanel(props: { initialQueryString: string }) {
+  const [queryString, setQueryString] = React.useState(props.initialQueryString);
+  const dispatch = useDispatch();
+  const query: string = useSelector((state: State) => state.plugins?.tp?.get('query'));
+  const queryResponse: Korrel8rGraphNeighboursResponse = useSelector((state: State) =>
+    state.plugins?.tp?.get('queryResponse'),
+  );
+
+  if (!query && queryString) {
+    dispatch(setQuery(queryString));
+  }
+
+  React.useEffect(() => {
+    if (!query) {
+      return;
+    }
+    const { request, abort } = getNeighborsGraph({ query });
+    request()
+      .then((response) => {
+        let existingNodes = response.nodes.filter((node) => node.count > 0);
+        existingNodes = response.nodes.map((node) => {
+          return {
+            class: node.class,
+            count: node.count,
+            queries: node.queries.filter((query) => query.count > 0),
+          };
+        });
+        const existingNodeNames = response.nodes.map((node) => node.class);
+        const existingEdges = response.edges
+          ? response.edges.filter(
+              (edge) =>
+                existingNodeNames.includes(edge.start) && existingNodeNames.includes(edge.goal),
+            )
+          : [];
+        dispatch(
+          setQueryResponse({
+            nodes: existingNodes,
+            edges: existingEdges,
+          }),
+        );
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
+    return () => {
+      abort();
+    };
+  }, [query, dispatch]);
+
+  const updateQuery = React.useCallback(() => {
+    dispatch(setQuery(queryString));
+  }, [dispatch, queryString]);
+
+  return (
+    <>
+      <FlexItem className="tp-plugin__panel-query-container">
+        <Title headingLevel="h2">Correlation Signals</Title>
+        <TextInputGroup className="tp-plugin__panel-query-input">
+          <TextArea
+            type="text"
+            id="queryString"
+            name="queryString"
+            value={queryString}
+            autoResize
+            onChange={(_event, value) => setQueryString(value)}
+          />
+        </TextInputGroup>
+        <Button isAriaDisabled={!queryString} onClick={updateQuery}>
+          Query
+        </Button>
+      </FlexItem>
+      <FlexItem className="tp-plugin__panel-topology-container" grow={{ default: 'grow' }}>
+        {queryResponse.nodes.length > 0 && queryResponse.edges.length > 0 ? (
```

Comment:
Created empty response state, generic error state, and korrel8r error state. Korrel8r error state shows the full error text since the backend responds with some helpful information such as "unmarshalling failed".